runtime/lua/vim/filetype/detect.lua

```diff
@@ -1256,17 +1256,16 @@ end
 -- 2. Check the first 1000 non-comment lines for LaTeX or ConTeXt keywords.
 -- 3. Default to "plain" or to g:tex_flavor, can be set in user's vimrc.
 function M.tex(path, bufnr)
-  local format = getlines(bufnr, 1):find('^%%&%s*(%a+)')
-  if format then
-    format = format:lower():gsub('pdf', '', 1)
-    if format == 'tex' then
-      return 'tex'
-    elseif format == 'plaintex' then
-      return 'plaintex'
-    end
+  local matched, idx, mstr = getlines(bufnr, 1):find('^%%&%s*(%a+)')
```

Comment:
Thanks for the PR! I would prefer to only change this line to `local _, _, format = getlines(bufnr, 1):find('^%%&%s*(%a+)')` though. The other changes such as directly returning the filetype were made deliberately to simplify the logic compared to Vim's implementation.

Reply:
The other changes you had unfortunately ended up changing the logic! If format didn't match in the first few lines, the fall back to `latex` is done by Vim, but not by the neovim you had before.

Just try whatever code you end up using on a file that has `%&preamble` in the first file. It should be detected as LaTeX (Vim does it).

Reply:
That's fine, but the tests (for more typical situations) -- which are taken verbatim from vim! -- must pass as well.

Reply:
fixed now